nibabel/cmdline/diff.py

```diff
@@ -0,0 +1,207 @@
+from __future__ import division, print_function, absolute_import
+
+import re
+import sys
+from collections import OrderedDict
+from optparse import OptionParser, Option
+
+import numpy as np
+
+import nibabel as nib
+import nibabel.cmdline.utils
+import hashlib
+import os
+
+
+def get_opt_parser():
+    # use module docstring for help output
+    p = OptionParser(
+        usage="%s [OPTIONS] [FILE ...]\n\n" % sys.argv[0] + __doc__,
+        version="%prog " + nib.__version__)
+
+    p.add_options([
+        Option("-v", "--verbose", action="count",
+               dest="verbose", default=0,
+               help="Make more noise.  Could be specified multiple times"),
+
+        Option("-H", "--header-fields",
+               dest="header_fields", default='all',
+               help="Header fields (comma separated) to be printed as well (if present)"),
+    ])
+
+    return p
+
+
+def are_values_different(*values):
+    """Generically compares values, returns true if different"""
+    value0 = values[0]
+    values = values[1:]  # to ensure that the first value isn't compared with itself
+
+    for value in values:
+        try:  # we sometimes don't want NaN values
+            if np.any(np.isnan(value0)) and np.any(np.isnan(value)):  # if they're both NaN
+                break
+            elif np.any(np.isnan(value0)) or np.any(np.isnan(value)):  # if only 1 is NaN
+                return True
+
+        except TypeError:
+            pass
+
+        if type(value0) != type(value):  # if types are different, then we consider them different
+            return True
+        elif isinstance(value0, np.ndarray):
+            return np.any(value0 != value)
+
+        elif value0 != value:
+            return True
+
+    return False
+
+
+def get_headers_diff(file_headers, names=None):
+    """Get difference between headers
+
+    Parameters
+    ----------
+    file_headers: list of actual headers (dicts) from files
+    names: list of header fields to test
+
+    Returns
+    -------
+    dict
+      str: list for each header field which differs, return list of
+      values per each file
+    """
+    difference = OrderedDict()
+    fields = names
+
+    if names is None:
+        fields = file_headers[0].keys()
+
+    # for each header field
+    for field in fields:
+        values = [header.get(field) for header in file_headers]  # get corresponding value
+
+        # if these values are different, store them in a dictionary
+        if are_values_different(*values):
+            difference[field] = values
+
+    return difference
+
+
+def get_data_diff(files):
+    """Get difference between md5 values
+
+        Parameters
+        ----------
+        files: list of actual files
+
+        Returns
+        -------
+        list
+          np.array: md5 values of respective files
+        """
+
+    md5sums = [
+        hashlib.md5(np.ascontiguousarray(nib.load(f).get_data(), dtype=np.float32)).hexdigest()
+        for f in files
+    ]
```

Comment:
Is there a reason you're using MD5 and not something more collision-resistant such as SHA256?